src/sync.cpp

```diff
@@ -149,10 +195,12 @@ void combinedFingerprint(FileFingerprint& ffp, FileSystemAccess& fsaccess, const
         if (fa->type == FILENODE)
         {
             FileFingerprint faFfp;
-            faFfp.genfingerprint(fa.get());
+            const auto index = fsaccess.lastpartlocal(const_cast<string*>(&path));
```

Comment:
Need to get the leaf name of the path here. This needs to be equivalent to `localname` of `LocalNode` so that the light-weight fingerprints become comparable.
Please let me know if there's a better more doing this.